Source/ASDisplayNode+Layout.mm

```diff
@@ -914,35 +916,40 @@ - (void)_assertSubnodeState
     return;
   }
 
-  NSArray *subnodes = [self subnodes];
-  NSArray *sublayouts = _calculatedDisplayNodeLayout.layout.sublayouts;
-
-  const auto currentSubnodes = [[NSHashTable alloc] initWithOptions:NSHashTableObjectPointerPersonality
-                                                     capacity:subnodes.count];
-  const auto layoutSubnodes  = [[NSHashTable alloc] initWithOptions:NSHashTableObjectPointerPersonality
-                                                     capacity:sublayouts.count];;
-  for (ASDisplayNode *subnode in subnodes) {
-    [currentSubnodes addObject:subnode];
-  }
-
-  for (ASLayout *sublayout in sublayouts) {
-    id <ASLayoutElement> layoutElement = sublayout.layoutElement;
-    ASDisplayNodeAssert([layoutElement isKindOfClass:[ASDisplayNode class]],
-                        @"All calculatedLayouts should be flattened and only contain nodes!");
-    [layoutSubnodes addObject:(ASDisplayNode *)layoutElement];
+  ASDN::MutexLocker l(__instanceLock__);
+  NSArray<ASLayout *> *sublayouts = _calculatedDisplayNodeLayout.layout.sublayouts;
+  unowned ASLayout *cSublayouts[sublayouts.count];
+  [sublayouts getObjects:cSublayouts range:NSMakeRange(0, AS_ARRAY_SIZE(cSublayouts))];
```

Comment:
Should `AS_ARRAY_SIZE(cSublayouts) == sublayouts.count` in this case and could drop `AS_ARRAY_SIZE` in this and also in the `if` check below?

Reply:
I wanted to avoid adding a new local variable or calling the count method multiple times. Since AS_ARRAY_SIZE is crazy fast I think it hits a good balance.